doc/whats_new.rst

```diff
@@ -85,6 +85,8 @@ Changelog
 
     - Add function :func:`mne.channels.get_builtin_montages` to list all built-in montages by `Clemens Brunner`_
 
+    - Add support for :class:`numpy.random.RandomState` argument to ``seed`` in :mod:`statistical clustering functions <mne.stats>` and better documentation of exact 1-sample tests by `Eric Larson`_
+
```

Comment:
No, `H0` was always correctly constructed. The docs were updated to make it clearer why sometimes `H0.size < n_permutations` (i.e., when an exact test is possible).

Technically this could slightly change results because it changes how the random state is used, but it hopefully shouldn't make a big difference to people (and if it does, people aren't doing enough permutations).